From e5ff23f36dd85a1ac26feb1191de8447c8d2009f Mon Sep 17 00:00:00 2001
From: Herman Chen <herman.chen@rock-chips.com>
Date: Tue, 10 Jul 2018 16:32:28 +0800
Subject: [PATCH] iommu/rockchip: export functions used in vcodec

When compile rk-vcodec as ko some unexported functions will fail the
compiling. This patch will fix this issue.

Change-Id: I8afcb4bc3858f0e9efa54e4996cdcb6a45bbe1e1
Signed-off-by: Herman Chen <herman.chen@rock-chips.com>
---
 drivers/iommu/rk-iovmm.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/iommu/rk-iovmm.c b/drivers/iommu/rk-iovmm.c
index 3dfa109e4361..087ccf60fe77 100644
--- a/drivers/iommu/rk-iovmm.c
+++ b/drivers/iommu/rk-iovmm.c
@@ -38,6 +38,7 @@ int rockchip_iovmm_invalidate_tlb(struct device *dev)
 
 	return ret;
 }
+EXPORT_SYMBOL(rockchip_iovmm_invalidate_tlb);
 
 void rockchip_iovmm_set_fault_handler(struct device *dev,
 				       rockchip_iommu_fault_handler_t handler)
@@ -46,6 +47,7 @@ void rockchip_iovmm_set_fault_handler(struct device *dev,
 
 	data->fault_handler = handler;
 }
+EXPORT_SYMBOL(rockchip_iovmm_set_fault_handler);
 
 int rockchip_iovmm_activate(struct device *dev)
 {
@@ -53,6 +55,7 @@ int rockchip_iovmm_activate(struct device *dev)
 
 	return iommu_attach_device(vmm->domain, dev);
 }
+EXPORT_SYMBOL(rockchip_iovmm_activate);
 
 void rockchip_iovmm_deactivate(struct device *dev)
 {
@@ -60,6 +63,7 @@ void rockchip_iovmm_deactivate(struct device *dev)
 
 	iommu_detach_device(vmm->domain, dev);
 }
+EXPORT_SYMBOL(rockchip_iovmm_deactivate);
 
 dma_addr_t rockchip_iovmm_map(struct device *dev,
 	struct scatterlist *sg, off_t offset, size_t size)
-- 
2.35.3

